src/core/SIP/SIPMessageBuffer.cs

```diff
@@ -36,11 +36,25 @@ public class SIPMessageBuffer
 
         private static ILogger logger = Log.Logger;
 
-        public string RawMessage;
+        public string RawMessage

```

Comment:
Now when there is extra overhead for encoding every time called. I think it's important to not missuse it like in SipTransport line 901 "SIPBadResponseInTraceEvent?.Invoke(localEndPoint, remoteEndPoint, sipMessageBuffer.RawMessage, sipValidationException.SIPErrorField, sipMessageBuffer.RawMessage);" where it's called twice. (I could not put any comment on that line for some reason).

Reply:
Yes another good point. But it's not crucial and it's better to confine this PR to the new feature. Makes it easier to review etc. 

There are a LOT of places in this library where strings are used inefficiently (particularly the SIPHeader class). But you'll find that under the hood the .NET runtimes do a very good job of optimising string operations. This library is typically doing network intensive operations and/or audio/video encoding both of which will be a much bigger bottle neck than processing small strings.

Happy to take subsequent PR's that optimise string processing.